test/e2e/addonmanagement_test.go

```diff
@@ -153,14 +153,24 @@ var _ = ginkgo.Describe("Enable addon management feature gate", ginkgo.Ordered,
 		gomega.Eventually(func() error {
 			_, err := hub.AddonClient.AddonV1alpha1().ManagedClusterAddOns(universalClusterName).Get(
 				context.TODO(), addOnName, metav1.GetOptions{})
+			if err == nil {
+				return fmt.Errorf("the managedClusterAddon %s should be deleted", addOnName)
+			}
+			if err != nil && !errors.IsNotFound(err) {
+				return err
+			}
+
+			// check works after addon is not found
+			works, err := hub.WorkClient.WorkV1().ManifestWorks(universalClusterName).List(
```

Comment:
I am thinking moving this check out of the eventually, to line 175, so if the addon is deleted, but there are works left, we make the test fail. and the print message will should which work left.

Reply:
done